evidence/doc.go

```diff
@@ -0,0 +1,36 @@
+/*
+Package evidence handles all evidence storage and gossiping from detection to block proposal.
+For the different types of evidence refer to the `evidence.go` file in the types package.
+
+## Gossiping
+
+The core functionality begins with the evidence reactor (see reactor.
+go) which operates both the sending and receiving of evidence.
+
+The `Receive` function takes a list of evidence and does the following:
+ 1. breaks it down into individual evidence if it is `Composite Evidence` (see types/evidence.
+go#ConflictingHeadersEvidence)
+ 2. checks that it does not already have the evidence stored
+ 3. verifies the evidence against the nodes state (see state/validation.go#VerifyEvidence)
+ 4. stores the evidence to a db and a concurrent list
+
+The gossiping of evidence is initiated when a peer is added which starts a go routine to broadcast currently
+uncommitted evidence at intervals of 60 seconds (set by the by broadcastEvidenceIntervalS).
+It uses a concurrent list to store the evidence and before sending verifies that each evidence is still valid in the
+sense that it has not exceeded the max evidence age and height which should be set to be equal to the "trusting
+period" (see types/params.go#EvidenceParams).
```

Comment:
Do you think we need more information regarding the setting of max evidence age and height (in regards to : https://github.com/tendermint/tendermint/issues/4670) @melekes

Reply:
no, it should be in ABCI / consensus params documentation, not here.